### PR TITLE
[CodeWhisperer] Add a document-changed threshold in CodeCoverageTracker

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if change is customer facing in the IDE.
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt [hunk 1]
```diff
@@ -91,6 +91,9 @@ abstract class CodeWhispererCodeCoverageTracker(
             LOG.debug { "event with isWholeTextReplaced flag: $event" }
             if (event.oldTimeStamp == 0L) return
         }
+
+        // We don't update totalToken if the document offset diff is larger than the threshold
+        if (event.newLength - event.oldLength >= DOCUMENT_CHANGE_THRESHOLD) return
```
Comment (andrewyuq) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt:96]: What's the reason behind this change?

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt:96]: the main reason I think is to block that loading scenario https://telemetry-externalprod.ide-toolkits.dev-tools.aws.dev/_plugin/kibana/goto/7205999dd5c2f77543c4d0bf82e5d370?security_tenant=global

since we can still see cases like, 
 totalToken = 10000, acceptedToken = 0
 totalToken = 5000, acceptedToken = 0

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt:96]: I suspect ^^ cases are a result of loading case.

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt:96]: Also I am thinking to  add a field "isCodeWhispererAutoSuggestion: boolean" in the `codewhisperer_percentage` metric as well, it will help us read the metric easier, know if the 0% is reasonable or not, IMO.

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt:96]: actually Lei already have set a threshold for VSC of length 20, but I would prefer larger number like hundred or thoushand

Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt:96]: don't x-link to internal resources please

Comment (Will-ShaoHua) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt:96]: ackk
